src/io/ioexternalbus.h

```diff
@@ -39,11 +42,24 @@ class IOExternalBus : public IOBase {
 protected:
     virtual void parameterChanged(unsigned) override;
 
+private slots:
+    void connectButtonTriggered();
+    // void sktConnected();
+    // void sktRead();
+
 private:
+    uint32_t ByteSize;
+    int32_t send_cmd(const uint32_t cmd, const uint32_t payload_size = 0, const char* payload = nullptr);
+    int32_t recv_cmd(cmd_header_t* cmd_header);
+    int32_t recv_payload(char* buff, const uint32_t payload_size);
+
     void updateAddress();
     Ui::IOExternalBus* m_ui = nullptr;
 
     std::vector<RegDesc> m_regDescs;
     std::vector<IOSymbol> m_extraSymbols;
+
+    XTcpSocket* tcpSocket = nullptr;
```

Comment:
Memory leak on this; best option would be to make it a unique ptr:
`std::unique_ptr<XTcpSocket> tcpSocket;`

Reply:
changed following your suggestion